Src/GoogleApis.Auth.DotNet4/OAuth2/ServiceAccountCredential.cs

```diff
@@ -66,12 +70,27 @@ public Initializer(string id)
                 : this(id, GoogleAuthConsts.TokenUrl) { }
 
             /// <summary>Constructs a new initializer using the given id and the token server URL.</summary>
-            public Initializer(string id, string tokenServerUrl) : base(tokenServerUrl)
+            public Initializer(string id, string tokenServerUrl)
+                : base(tokenServerUrl)
             {
                 Id = id;
                 Scopes = new List<string>();
             }
 
+            /// <summary>Constructs a new initializer using the given id and the private key in the ClientCredentialParameters.</summary>
+            public Initializer(ClientCredentialParameters clientCredentialParameters)
+                : this(clientCredentialParameters.ClientEmail, GoogleAuthConsts.TokenUrl)
+            {
+                Utilities.ThrowIfNullOrEmpty(clientCredentialParameters.ClientEmail, "ClientId");
+                Utilities.ThrowIfNullOrEmpty(clientCredentialParameters.Base64PrivateKey, "ClientSecret");
+
+                var privateKeyBytes = Convert.FromBase64String(clientCredentialParameters.Base64PrivateKey);
+                RsaPrivateCrtKeyParameters crtParameters = (RsaPrivateCrtKeyParameters)PrivateKeyFactory.CreateKey(privateKeyBytes);
+                RSAParameters rsaParameters = DotNetUtilities.ToRSAParameters(crtParameters);

```

Comment:
To explain this: This change also adds support for the new JSON format for service accounts. These JSON files contain the private key in PKCS8 format.

As there seems to be no way library to read PKCS8 keys in .NET, bouncyCastles is used to read the key. The eventual goal is to create an RSACryptoServiceProvider object (that is required by the old service credential code). The code in this method is basically trying to convert JSON key to RSACryptoServiceProvider (similarly to FromCertificate(X509Certificate2) )